soc/bflb/bl61x/soc.c

```diff
@@ -19,27 +19,91 @@
 #include <hbn_reg.h>
 #include <pds_reg.h>
 
+#define SYSMAP_BASE 0xEFFFF000UL
+#define SYSMAP_BASE_OFFSET       (12)
+#define SYSMAP_ATTR_STRONG_ORDER BIT(4)
+#define SYSMAP_ATTR_CACHE_ABLE   BIT(3)
+#define SYSMAP_ATTR_BUFFER_ABLE  BIT(2)
+#define SYSMAP_ADDR_OFFSET       0x0
+#define SYSMAP_FLAGS_OFFSET      0x4
+#define SYSMAP_ENTRY_OFFSET      0x8
+
+/* Initialize memory regions */
+void system_sysmap_init(void)
+{
+	uintptr_t sysmap_base = SYSMAP_BASE;
+
+	/* 1. 0x00000000~0x62FC0000: Strong-Order, Non-Cacheable, Non-Bufferable */
+	sys_write32(BL616_OCRAM_BUSREMAP_CACHEABLE_BASE >> SYSMAP_BASE_OFFSET,
+		    (sysmap_base + SYSMAP_ADDR_OFFSET));
+	sys_write32(SYSMAP_ATTR_STRONG_ORDER, (sysmap_base + SYSMAP_FLAGS_OFFSET));
+	sysmap_base += SYSMAP_ENTRY_OFFSET;
+
+	/* 2. ocram 0x62FC0000~0x63010000: Weak-Order, Cacheable, Bufferable */
+	sys_write32(BL616_WRAM_BUSREMAP_CACHEABLE_BASE >> SYSMAP_BASE_OFFSET,
+		    (sysmap_base + SYSMAP_ADDR_OFFSET));
+	sys_write32(SYSMAP_ATTR_CACHE_ABLE | SYSMAP_ATTR_BUFFER_ABLE,
+		    (sysmap_base + SYSMAP_FLAGS_OFFSET));
+	sysmap_base += SYSMAP_ENTRY_OFFSET;
+
+	/* 3. wram 0x63010000~0x63038000: Weak-Order, Cacheable, Bufferable */
+	sys_write32(0x63038000U >> SYSMAP_BASE_OFFSET, (sysmap_base + SYSMAP_ADDR_OFFSET));
+	sys_write32(SYSMAP_ATTR_CACHE_ABLE | SYSMAP_ATTR_BUFFER_ABLE,
+		    (sysmap_base + SYSMAP_FLAGS_OFFSET));
+	sysmap_base += SYSMAP_ENTRY_OFFSET;
```

Comment:
This is matching `sram1`:
```
                sram1: memory@63010000 {
                        compatible = "zephyr,memory-region", "mmio-sram";
                        reg = <0x63010000 DT_SIZE_K(160)>;
                        zephyr,memory-region = "ITCM";
                };
```

Is this a TODO to use the values from devicetree or is it considered more like vendor-provided data?

I am wondering about the long-term plan, not asking about this PR.

Reply:
done